src/__tests__/container.test.ts

```diff
@@ -588,6 +590,184 @@ describe('container', () => {
       expect(theAnswer()).toBe(42)
     })
   })
+
+  describe('strict mode', () => {
+    describe('lifetime mismatch check', () => {
+      it('allows longer lifetime modules to depend on shorter lifetime dependencies by default', () => {
+        const container = createContainer()
+        container.register({
+          first: asFunction((cradle: any) => cradle.second, {
+            lifetime: Lifetime.SCOPED,
+          }),
+          second: asFunction(() => 'hah'),
+        })
+
+        expect(container.resolve('first')).toBe('hah')
+      })
+
+      it('throws an AwilixResolutionError when longer lifetime modules depend on shorter lifetime dependencies and strict is set', () => {
+        const container = createContainer({
+          strict: true,
+        })
+        container.register({
+          first: asFunction((cradle: any) => cradle.second, {
+            lifetime: Lifetime.SCOPED,
+          }),
+          second: asFunction(() => 'hah'),
+        })
+
+        const err = throws(() => container.resolve('first'))
+        expect(err.message).toContain('first -> second')
+        expect(err.message).toContain(
+          "Dependency 'second' has a shorter lifetime than its ancestor: 'first'",
+        )
+      })
+
+      it('does not throw an error when an injector proxy is used and strict is set', () => {
+        const container = createContainer({
+          strict: true,
+        })
+        container.register({
+          first: asFunction((cradle: any) => cradle.injected, {
+            lifetime: Lifetime.SCOPED,
+          }).inject(() => ({ injected: 'hah' })),
```

Comment:
What if we also add `injected` as a registration as well as use the proxy? It won't trigger the check right? As in, this test would still pass?

Reply:
correct, added the test